src/boot/efi/stub.c

```diff
@@ -222,7 +237,18 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
 
         export_variables(loaded_image);
 
-        (void) pack_cpio(loaded_image,
+        loaded_image_root = LibOpenRoot(loaded_image->DeviceHandle);
+        if (!loaded_image_root)
+                return log_error_status_stall(EFI_LOAD_ERROR, L"Unable to open root directory.");
+        loaded_image_path = DevicePathToStr(loaded_image->FilePath);
+        if (!loaded_image_path)
+                return log_oom();
+        initrd_dropin_dir = PoolPrint(L"%s.extra.d", loaded_image_path);
```

Comment:
You can pass `loaded_image->FilePath` directly to PoolPrint by using the `%D` format specifier.